apps/desktop-wallet/src/storage/prices/pricesActions.ts

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { CHART_DATE_FORMAT, TOKENS_QUERY_LIMIT } from '@alephium/shared'
+import { createAsyncThunk } from '@reduxjs/toolkit'
+import dayjs from 'dayjs'
+import { chunk } from 'lodash'
+
+import client from '@/api/client'
+import { TokenHistoricalPrice, TokenPriceEntity, TokenPriceHistoryEntity } from '@/types/price'
+import { isFulfilled } from '@/utils/misc'
+
+export const syncTokenCurrentPrices = createAsyncThunk(
+  'assets/syncTokenCurrentPrices',
+  async ({ verifiedFungibleTokenSymbols, currency }: { verifiedFungibleTokenSymbols: string[]; currency: string }) => {
+    let tokenPrices = [] as TokenPriceEntity[]
+
+    const promiseResults = await Promise.allSettled(
```

Comment:
We cannot use `Promise.all` here because if one call for a token price fails, then all will fail.